### PR TITLE
Update testing to new approach

### LINKED ISSUE #11: Update testing to new approach
Update tests to use Cloud Build and make sure tests all pass.

### PR DESCRIPTION
**Pull request by kopachevsky**
Fixes #11

### CODE REVIEW

File: build/lint.cloudbuild.yaml [hunk 1]
```diff
@@ -11,3 +11,14 @@
...
+steps:
+  - name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+    id: 'lint'
+    args: ['/usr/local/bin/test_lint.sh']
+tags:
+  - 'ci'
+  - 'lint'
+substitutions:
+  _DOCKER_IMAGE_DEVELOPER_TOOLS: 'cft/developer-tools'
+  _DOCKER_TAG_VERSION_DEVELOPER_TOOLS: '0.4.6'
```
Comment (paulpalamarchuk) [inline build/lint.cloudbuild.yaml]: Why do you use different docker image versions for lint and integration tests?

File: test/setup/iam.tf [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  int_required_roles = [
+    "roles/owner",
+    "roles/dataflow.admin",
+    "roles/dataflow.developer",
+    "roles/dataflow.viewer",
```
Comment (paulpalamarchuk) [inline test/setup/iam.tf]: Do we need "roles/dataflow.developer" and "roles/dataflow.viewer"?

File: test/setup/versions.tf [hunk 3]
```diff
@@ -13,3 +13,15 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+terraform {
+  required_version = ">= 0.12"
+}
+
+provider "google" {
+  version = "~> 2.19.0"
```
Comment (paulpalamarchuk) [inline test/setup/versions.tf]: Is that necessary?

File: Makefile [hunk 4]
```diff
@@ -18,147 +18,68 @@
 # Make will use bash instead of sh
 SHELL := /usr/bin/env bash
 
-# Docker build config variables
-CREDENTIALS_PATH 			?= /cft/workdir/credentials.json
-DOCKER_ORG 				:= gcr.io/cloud-foundation-cicd
-DOCKER_TAG_BASE_KITCHEN_TERRAFORM 	?= 2.3.0
-DOCKER_REPO_BASE_KITCHEN_TERRAFORM 	:= ${DOCKER_ORG}/cft/kitchen-terraform:${DOCKER_TAG_BASE_KITCHEN_TERRAFORM}
+DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0.5.2
```
Comment (aaron-lane) [inline Makefile]: This can be simplified.

```suggestion
DOCKER_TAG_VERSION_DEVELOPER_TOOLS := 0
```

File: build/int.cloudbuild.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do verify']
+  - id: destroy
+    name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+    args: ['/bin/bash', '-c', 'source /usr/local/bin/task_helper_functions.sh && kitchen_do destroy']
+tags:
+  - 'ci'
+  - 'integration'
+substitutions:
+  _DOCKER_IMAGE_DEVELOPER_TOOLS: 'cft/developer-tools'
+  _DOCKER_TAG_VERSION_DEVELOPER_TOOLS: '0.5.2'
```
Comment (aaron-lane) [inline build/int.cloudbuild.yaml]:
```suggestion
  _DOCKER_TAG_VERSION_DEVELOPER_TOOLS: '0'
```

File: build/lint.cloudbuild.yaml [hunk 6]
```diff
@@ -11,3 +11,14 @@
...
+steps:
+  - name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+    id: 'lint'
+    args: ['/usr/local/bin/test_lint.sh']
+tags:
+  - 'ci'
+  - 'lint'
+substitutions:
+  _DOCKER_IMAGE_DEVELOPER_TOOLS: 'cft/developer-tools'
+  _DOCKER_TAG_VERSION_DEVELOPER_TOOLS: '0.5.2'
```
Comment (aaron-lane) [inline build/lint.cloudbuild.yaml]:
```suggestion
  _DOCKER_TAG_VERSION_DEVELOPER_TOOLS: '0'
```

Comment (kopachevsky) [reply build/lint.cloudbuild.yaml]: after update have issue with Project factory
```
on .terraform/modules/project-ci-dataflow/terraform-google-modules-terraform-google-project-factory-f93d3cd/modules/core_project_factory/main.tf line 165, in resource "google_project_services" "project_services_authority":
165: resource "google_project_services" "project_services_authority" {

The provider provider.google does not support resource type
"google_project_services".
```

Comment (kopachevsky) [reply build/lint.cloudbuild.yaml]: it's actually not depends on image version, will discuss on a today call

File: modules/dataflow_bucket/versions.tf [hunk 7]
```diff
@@ -17,3 +17,11 @@
 terraform {
   required_version = ">= 0.12"
 }
+
+provider "google" {
```
Comment (morgante) [inline modules/dataflow_bucket/versions.tf]: Providers should never be configured within a module. Instead, you can set version constraints on the https://www.terraform.io/docs/configuration/modules.html#provider-version-constraints-in-modules

File: test/fixtures/simple_example/versions.tf [hunk 8]
```diff
@@ -17,3 +17,11 @@
 terraform {
   required_version = ">= 0.12"
 }
+
+provider "google" {
```
Comment (morgante) [inline test/fixtures/simple_example/versions.tf]: Put these on the example, not the fixture.

File: modules/dataflow_bucket/versions.tf (hunk 7 above)
Comment (kopachevsky) [reply modules/dataflow_bucket/versions.tf]: done

File: test/fixtures/simple_example/versions.tf (hunk 8 above)
Comment (kopachevsky) [reply test/fixtures/simple_example/versions.tf]: done

**Pull request merged**